network/common/src/main/java/org/apache/spark/network/util/JavaUtils.java

```diff
@@ -121,4 +123,68 @@ private static boolean isSymlink(File file) throws IOException {
     }
     return !fileInCanonicalDir.getCanonicalFile().equals(fileInCanonicalDir.getAbsoluteFile());
   }
+
+  private static ImmutableMap<String, TimeUnit> timeSuffixes = 
+    ImmutableMap.<String, TimeUnit>builder()
+      .put("us", TimeUnit.MICROSECONDS)
+      .put("ms", TimeUnit.MILLISECONDS)
+      .put("s", TimeUnit.SECONDS)
+      .put("m", TimeUnit.MINUTES)
+      .put("min", TimeUnit.MINUTES)
+      .put("h", TimeUnit.HOURS)
+      .put("d", TimeUnit.DAYS)
+      .build();
+
+  /**
+   * Convert a passed time string (e.g. 50s, 100ms, or 250us) to a time count for
+   * internal use. If no suffix is provided a direct conversion is attempted.
+   */
+  private static long parseTimeString(String str, TimeUnit unit) {
+    String lower = str.toLowerCase().trim();
+    
+    try {
+      String suffix = "";
+      for (String tail: timeSuffixes.keySet()) {
+        int charIdx = lower.length() - tail.length() - 1;
+        if (lower.endsWith(tail) && (charIdx >= 0 && Character.isDigit(lower.charAt(charIdx)))) {
+          suffix = tail;
+          break;
+        }
+      }
```

Comment:
Using regexes this would be pretty short.

    Matcher m = Pattern.compile("([0-9]+)([a-z]+)?");
    if (m.matches()) {
       long val = Long.parseLong(m.group(1));
       String suffix = m.group(2);
    }

Or something along those lines.